File: Baekjoon/No10828/Solution.java
package Baekjoon.No10828;

import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>(); // 스택 선언. 타입에 따라 지정하여 선언.
        int num = sc.nextInt(); // 명령개수

        // 명령개수만큼 반복
        for (int i = 0; i < num; i++) {
            String order = sc.next();

            // 다섯가지 명령에 대해 분리
            switch (order) {
                case "push":
                    stack.push(sc.nextInt());
                    break;

                case "pop":
                    if (stack.empty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.pop());
                    }
                    break;

                case "size":
                    System.out.println(stack.size());
                    break;

                case "empty":
                    if (stack.empty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;

                case "top":
                    if (stack.empty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack.peek());
                    }
                    break;
            }
        }

        sc.close();
    }
}

File: Baekjoon/No10828/Solution.cpp
#include <iostream>
#include <stack>
using namespace std;

int num, temp;
string order;
stack<int> stk;

void func()
{
    cin >> order;

    if (order == "push")
    {
        cin >> temp;
        stk.push(temp);
    }
    else if (order == "pop")
    {
        if (stk.empty())
            cout << -1 << endl;
        else
        {
            cout << stk.top() << endl;
            stk.pop();
        }
    }
    else if (order == "size")
        cout << stk.size() << endl;
    else if (order == "empty")
    {
        if (stk.empty())
            cout << 1 << endl;
        else
            cout << 0 << endl;
    }
    else
    {
        if (stk.empty())
            cout << -1 << endl;
        else
            cout << stk.top() << endl;
    }
}

int main()
{
    cin >> num;

    for (int i = 0; i < num; i++)
    {
        func();
    }
}